benchmarks/algorithm/rotate.cpp

```diff
@@ -0,0 +1,74 @@
+#include <nanorange/algorithm/rotate.hpp>
+
+#include <algorithm>
+#include <utility>
+#include <vector>
+
+#include <nanorange/algorithm/generate.hpp>
+
+#include <benchmark/benchmark.h>
+
+namespace {
+
+void set_interesting_sizes(benchmark::internal::Benchmark* bench)
+{
+    std::pair<int, int> cases[]{
+        {1, 999},           // Special case for std
+        {100, 900},         //
+        {400, 600},         //
+        {500, 500},         // Special case for std
+        {700, 300},         //
+        {999, 1},           // Special case for std
+        {700'000, 350'000}, // Big gcd
+        {30'000, 16'127},   // Coprime
+    };
+
+    for (auto c : cases) {
+        bench->Args({c.first, c.second});
+    }
+}
+
+template <typename T>
+struct type_t {
+    using type = T;
+};
+
+auto generator(type_t<int>)
+{
+    return [x = 0]() mutable { return x++; };
+}
+
+template <typename F, typename T>
+void rotate_random_access(benchmark::State& state)
+{
+    const std::size_t lhs = static_cast<std::size_t>(state.range(0)),
+                      rhs = static_cast<std::size_t>(state.range(1));
+
+    std::vector<T> v(lhs + rhs);
+    nano::generate(v, generator(type_t<T>{}));
```

Comment:
Is there a need for `generator` and `type_t`, or could you just call `nano::generate` with a lambda directly?

Reply:
I wanted to have an extension point to do strings or things like that, where maybe the price of move would be bigger.

Reply:
Added strings.